src/Parsers/Access/ParserCreateUserQuery.cpp

```diff
@@ -141,6 +154,34 @@ namespace
             }
 
             auth_data = AuthenticationData{*type};
+            if (auth_data.getType() == AuthenticationType::SHA256_PASSWORD)
+            {
+                if (!parsed_salt.empty())
+                {
+                    auth_data.setSalt(parsed_salt);
+                }
+                else if (expect_password)
+                {
+#if USE_SSL
+                    ///generate and add salt here
+                    ///random generator FIPS complaint
+                    uint8_t key[32];
+                    RAND_bytes(key, sizeof(key));
```

Comment:
reinterpret_cast is ok between char * and uint8_t * (exception from strict aliasing, so the extra copy is unnecessary)